### DISCUSSION TITLE
Excalidraw Docker Extension

### DISCUSSION THREAD

**Discussion by ivan-curkovic** [excalidraw/excalidraw#6317] [Show and tell]
Hi,
I created Excalidraw Docker Extension for fun. 
This allows Docker Desktop users to install Excalidraw with one click.

🔗Repository: https://github.com/ivan-curkovic/excalidraw-docker-extension
🐋Link to open in Docker Desktop: https://open.docker.com/extensions/marketplace?extensionId=ivancurkovic046/excalidraw-docker-extension

Would like to give it over to the Excalidraw organization. What do you think?

**Comment by Lewiscowles1986** [comment]
Hey Ivan,

looks really good. I Just made a branch for [MailHog](https://github.com/CODESIGN2/mailhog-docker/pull/11) kicking the tyres, and your extension was both installed and an inspiration for integrating within Docker Desktop.

I'm likely just kicking the tyres of what is possible here, but did you encounter anything difficult or strange, and what do you think about the architecture of Docker-Desktop Extensions?

**Reply by ivan-curkovic** [reply]
Hi Lewis, great to see new ideas and possibilities. Looking forward to trying your extension. Will you publish it to the marketplace? 

My extension was quite trivial and building it was easy, what was your experience?

**Reply by Lewiscowles1986** [reply]
So mine isn't done yet (although it works). It can be installed via cli, despite not being available on the docker store. (I don't think I'll maintain it).

```
docker pull cd2team/mailhog-docker-desktop-extension:snapshot
docker extension install cd2team/mailhog-docker-desktop-extension:snapshot
```

I'd like to see if I can avoid exposing the web-port, or allow the user UI to change the docker image ports so that it's less fragile (I don't like relying on users having ports open).

In my notes I have taken note that you were able to use an iframe, rather than keep an app as an SPA, which is very cool.

**Reply by Lewiscowles1986** [reply]
I Also looked at the localstack one, but it was more fragile still. Has a note on a page I didn't have to view to install it, that suggests you need to change docker-desktop settings to use it 🤷 

It also does not start localstack by default, whereas the user story I imagined creating for, was someone that might struggle with CLI past install
